SQL: AWS Glue (Postgres JDBC) `Expected command status BEGIN, got START TRANSACTION`

<!-- Please search existing issues to avoid creating duplicates. -->

<!--
For the Gel Version: run `gel query 'select sys::get_version_as_str()'` from your project directory (or run `select sys::get_version_as_str();` in the Gel interactive shell).
For the Gel CLI Version: Run `gel --version` from anywhere
-->

- Gel Version: 6.7
- Gel CLI Version: n/a
- OS Version: amazon linux


Background: Normally would want postgres logical replication for ETL, but since that's not supported we're using Glue so that we can do everything through dumb queries

Steps to Reproduce:

1. Connect to Gel's SQL protocol support using AWS Glue
2. `org.postgresql.util.PSQLException: Expected command status BEGIN, got START TRANSACTION.`


Full exception from the glue job

```
Py4JJavaError - An error occurred while calling o270.collectToPython. : org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (10.0.138.110 executor 1): org.postgresql.util.PSQLException: Expected command status BEGIN, got START TRANSACTION. at 
org.postgresql.core.v3.QueryExecutorImpl$1.handleCommandStatus(QueryExecutorImpl.java:642) at org.postgresql.core.v3.QueryExecutorImpl.interpretCommandStatus(QueryExecutorImpl.java:2770) at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2360) at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:371) at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:502) at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:419) at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:194) at org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:137) at org.apache.spark.sql.jdbc.glue.JDBCRDD.compute(GlueJDBCSource.scala:148) at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:378) at org.apache.spark.rdd.RDD.iterator(RDD.scala:333) at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:61) at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:378) at org.apache.spark.rdd.RDD.iterator(RDD.scala:333) at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:61) at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:378) at org.apache.spark.rdd.RDD.iterator(RDD.scala:333) at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:61) at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:378) at org.apache.spark.rdd.RDD.iterator(RDD.scala:333) at org.apache.spark.shuffle.ShuffleWriteProcessor.doWrite(ShuffleWriteProcessor.scala:45) at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:69) at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:104) at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:54) at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:174) at org.apache.spark.scheduler.Task.run(Task.scala:152) at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:632) at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64) at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61) at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:96) at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:635) at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) at java.base/java.lang.Thread.run(Thread.java:840) Driver stacktrace: at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:3083) at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:3019) at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:3018) at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62) at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55) at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49) at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:3018) at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1324) at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1324) at scala.Option.foreach(Option.scala:407) at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1324) at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3301) at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:3235) at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:3224) at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49) at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:1047) at org.apache.spark.SparkContext.runJob(SparkContext.scala:2499) at org.apache.spark.SparkContext.runJob(SparkContext.scala:2520) at org.apache.spark.SparkContext.runJob(SparkContext.scala:2539) at org.apache.spark.SparkContext.runJob(SparkContext.scala:2564) at org.apache.spark.rdd.RDD.$anonfun$collect$1(RDD.scala:1065) at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:152) at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:113) at org.apache.spark.rdd.RDD.withScope(RDD.scala:426) at org.apache.spark.rdd.RDD.collect(RDD.scala:1064) at org.apache.spark.sql.execution.SparkPlan.executeCollect(SparkPlan.scala:516) at org.apache.spark.sql.Dataset.$anonfun$collectToPython$1(Dataset.scala:4222) at org.apache.spark.sql.Dataset.$anonfun$withAction$2(Dataset.scala:4392) at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:714) at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:4390) at org.apache.spark.sql.catalyst.QueryPlanningTracker$.withTracker(QueryPlanningTracker.scala:108) at org.apache.spark.sql.execution.SQLExecution$.withTracker(SQLExecution.scala:384) at org.apache.spark.sql.execution.SQLExecution$.executeQuery$1(SQLExecution.scala:157) at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$10(SQLExecution.scala:220) at org.apache.spark.sql.catalyst.QueryPlanningTracker$.withTracker(QueryPlanningTracker.scala:108) at org.apache.spark.sql.execution.SQLExecution$.withTracker(SQLExecution.scala:384) at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$9(SQLExecution.scala:220) at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:405) at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:219) at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:901) at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:83) at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:74) at org.apache.spark.sql.Dataset.withAction(Dataset.scala:4390) at org.apache.spark.sql.Dataset.collectToPython(Dataset.scala:4219) at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.base/java.lang.reflect.Method.invoke(Method.java:569) at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244) at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374) at py4j.Gateway.invoke(Gateway.java:282) at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132) at py4j.commands.CallCommand.execute(CallCommand.java:79) at py4j.GatewayConnection.run(GatewayConnection.java:238) at java.base/java.lang.Thread.run(Thread.java:840) Caused by: org.postgresql.util.PSQLException: Expected command status BEGIN, got START TRANSACTION. at org.postgresql.core.v3.QueryExecutorImpl$1.handleCommandStatus(QueryExecutorImpl.java:642) at org.postgresql.core.v3.QueryExecutorImpl.interpretCommandStatus(QueryExecutorImpl.java:2770) at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2360) at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:371) at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:502) at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:419) at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:194) at org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:137) at org.apache.spark.sql.jdbc.glue.JDBCRDD.compute(GlueJDBCSource.scala:148) at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:378) at org.apache.spark.rdd.RDD.iterator(RDD.scala:333) at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:61) at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:378) at org.apache.spark.rdd.RDD.iterator(RDD.scala:333) at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:61) at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:378) at org.apache.spark.rdd.RDD.iterator(RDD.scala:333) at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:61) at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:378) at org.apache.spark.rdd.RDD.iterator(RDD.scala:333) at org.apache.spark.shuffle.ShuffleWriteProcessor.doWrite(ShuffleWriteProcessor.scala:45) at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:69) at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:104) at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:54) at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:174) at org.apache.spark.scheduler.Task.run(Task.scala:152) at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:632) at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64) at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61) at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:96) at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:635) at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ... 1 more
```

--- a/edb/server/compiler/sql.py
+++ b/edb/server/compiler/sql.py
@@ -317,7 +317,11 @@ def _compile_sql(
         elif isinstance(stmt, (pgast.BeginStmt, pgast.StartStmt)):
             unit.tx_action = dbstate.TxAction.START
             unit.command_complete_tag = dbstate.TagPlain(
-                tag=b"START TRANSACTION"
+                tag=(
+                    b"START TRANSACTION"
+                    if isinstance(stmt, pgast.StartStmt)
+                    else b"BEGIN"
+                )
             )
         elif isinstance(stmt, pgast.CommitStmt):
             unit.tx_action = dbstate.TxAction.COMMIT

--- a/tests/test_sql_query.py
+++ b/tests/test_sql_query.py
@@ -3478,6 +3478,20 @@ class TestSQLQueryNonTransactional(tb.SQLQueryTestCase):
         v2 = await self.scon.fetchval("SHOW transaction_isolation")
         self.assertNotEqual(v1, v2)
 
+    async def test_sql_transaction_03(self):
+        self.assertEqual(
+            await self.scon.execute("BEGIN"),
+            "BEGIN",
+        )
+        await self.scon.execute("ROLLBACK")
+
+    async def test_sql_transaction_04(self):
+        self.assertEqual(
+            await self.scon.execute("START TRANSACTION"),
+            "START TRANSACTION",
+        )
+        await self.scon.execute("ROLLBACK")
+
     async def test_sql_query_error_11(self):
         # extended query protocol
         with self.assertRaisesRegex(

